cni/network/network.go

```diff
@@ -1404,3 +1438,13 @@ func convertCniResultToInterfaceInfo(result *cniTypesCurr.Result) network.Interf
 
 	return interfaceInfo
 }
+
+func findDefaultInterface(ipamAddResult IPAMAddResult) (int, error) {
```

Comment:
maybe return interface info instead of index

Reply:
If that is something that we want to do instead it can be done. Having the index is more lightweight than passing an entire struct through the flow. Would rather have the next PR change interfaceInfo into a map and then we no longer need to search for infra nic.